pkg/querier/tenantfederation/merge_queryable_test.go

```diff
@@ -195,241 +213,465 @@ func (m mockTenantQuerier) LabelNames() ([]string, storage.Warnings, error) {
+				// 		name:                "should return only label values for team-a and team-c tenants when there is a not-equals matcher for team-b tenant",
+				// 		labelName:           defaultTenantLabel,
+				// 		matchers:            []*labels.Matcher{{Name: defaultTenantLabel, Value: "team-b", Type: labels.MatchNotEqual}},
+				// 		expectedLabelValues: []string{"team-a", "team-c"},
+				// 	},
+				// 	{
+				// 		name:                "should return only label values for team-b tenant when there is an equals matcher for team-b tenant",
+				// 		labelName:           defaultTenantLabel,
+				// 		matchers:            []*labels.Matcher{{Name: defaultTenantLabel, Value: "team-b", Type: labels.MatchEqual}},
+				// 		expectedLabelValues: []string{"team-b"},
+				// 	},
+			},
+		},
+		{
+			mergeQueryableScenario: threeTenantsWithDefaultTenantIDScenario,
+			labelValuesTestCases: []labelValuesTestCase{
+				{
+					name:                "should return all label values for instance when no matchers are provided",
+					labelName:           "instance",
+					expectedLabelValues: []string{"host1", "host2.team-a", "host2.team-b", "host2.team-c"},
+				},
+				{
+					name:                "should return all tenant values for __tenant_id__ label name",
+					labelName:           defaultTenantLabel,
+					expectedLabelValues: []string{"team-a", "team-b", "team-c"},
+				},
+				{
+					name:                "should return the original value for the revised tenant labelname when no matchers are provided",
+					labelName:           originalDefaultTenantLabel,
+					expectedLabelValues: []string{"original-value"},
+				},
+				{
+					name:             "with equals matcher for __original_tenant_id__ label it should error as matchers aren't implemented for LabelValues yet",
+					matchers:         []*labels.Matcher{{Name: originalDefaultTenantLabel, Value: "original-value", Type: labels.MatchEqual}},
+					labelName:        originalDefaultTenantLabel,
+					expectedQueryErr: errMatchersNotImplemented,
+				},
+			},
+		},
+		// TODO(jdb): Should these errors be warnings and errors be present?
+		// Currently they are not returned by the LabelValues.
+		// {
+		// 	mergeQueryableScenario: threeTenantsWithWarningsScenario,
+		// 	labelValuesTestCases: []labelValuesTestCase{{
+		// 		name:                "warnings from all merged queryables should be returned",
+		// 		labelName:           defaultTenantLabel,
+		// 		expectedLabelValues: []string{"team-a", "team-b", "team-c"},
+		// 		expectedWarnings: []string{
+		// 			`warning querying tenant_id team-b: don't like them`,
+		// 			`warning querying tenant_id team-c: out of office`,
+		// 		},
+		// 	}},
+		// },
+		// {
+		// 	mergeQueryableScenario: threeTenantsWithErrorScenario,
+		// 	labelValuesTestCases: []labelValuesTestCase{{
+		// 		name:             "any error encountered with a single tenant should be returned",
+		// 		expectedQueryErr: errors.New("error querying tenant_id team-b: failure xyz"),
```

Comment:
You are expecting the wrong error:

                                expected: *errors.withStack(error querying tenant_id team-b: failure xyz)
                                actual  : *errors.errorString(&errors.errorString{s:"error querying tenant_id team-b: failure xyz"})